src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs

```diff
@@ -167,11 +168,24 @@ public static SBMessage AmqpMessageToSBMessage(AmqpMessage amqpMessage)
             }
 
             SBMessage sbMessage;
-            if (amqpMessage.BodyType == SectionFlag.AmqpValue && amqpMessage.ValueBody.Value != null)
+
+            if ((amqpMessage.BodyType & SectionFlag.AmqpValue) != 0
+                && amqpMessage.ValueBody.Value != null)
             {
                 var byteArrayValue = (byte[])amqpMessage.ValueBody.Value;
                 sbMessage = new SBMessage(byteArrayValue);
             }
+            else if ((amqpMessage.BodyType & SectionFlag.Data) != 0
+                && amqpMessage.DataBody?.Count() > 0)
```

Comment:
>DataBody?.Count() [](start = 31, length = 17)

Avoid Linq where possible, just add a check to DataBody! = null., here. the Ienumerable will will make sure that it is entered only if count > 0